### PR TITLE
[IMPROVEMENT] Hide the drawing toolbar while drawing

### LINKED ISSUE #2066: [IMPROVEMENT] Hide the drawing toolbar while drawing.
## Description
It's good if we hide the drawing toolbar while drawing which give users more area to draw.

### PR DESCRIPTION
**Pull request by kareemhamdy100**
@RocketChat/android

Closes #2066

#### Screenshots or GIF for the change:
device-2019-03-04-022641
version 2 : 
version2

### CODE REVIEW

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt [hunk 1]
```diff
@@ -41,10 +41,14 @@ class DrawingActivity : DaggerAppCompatActivity(), DrawView {
 
     override fun showWrongProcessingMessage() {
         Toast.makeText(this, getText(R.string.msg_wrong_processing_draw_image), Toast.LENGTH_SHORT)
-            .show()
+                .show()
```
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt [hunk 2]
```diff
@@ -102,65 +106,72 @@ class DrawingActivity : DaggerAppCompatActivity(), DrawView {
         }
     }
 
-    private fun toggleDrawTools(view: View, showView: Boolean = true) {
-        if (showView) {
+    private fun toggleDrawTools(view: View, showView: Boolean = true, all_show: Boolean = true) {
+        if (showView)
```
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong code style.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt [hunk 3]
```diff
@@ -102,65 +106,72 @@ class DrawingActivity : DaggerAppCompatActivity(), DrawView {
...
             view.animate().translationY((0).toPx)
-        } else {
+         else
             view.animate().translationY((56).toPx)
-        }
     }
 
+    private fun toggleCompleteDrawTools(view: View,showView: Boolean = true) {
+        if (view.translationY == (112).toPx && showView)
```
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong code style.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt [hunk 4]
```diff
@@ -102,65 +106,72 @@ class DrawingActivity : DaggerAppCompatActivity(), DrawView {
...
+        else
+            view.animate().translationY((112).toPx)
+    }
+
+
     private fun colorSelector() {
         image_color_red.setOnClickListener {
             custom_draw_view.setColor(
-                ResourcesCompat.getColor(resources, R.color.color_red, null)
+                    ResourcesCompat.getColor(resources, R.color.color_red, null)
```
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt [hunk 5]
```diff
@@ -102,65 +106,72 @@ class DrawingActivity : DaggerAppCompatActivity(), DrawView {
...
             scaleColorView(image_color_red)
         }
 
         image_color_yellow.setOnClickListener {
             custom_draw_view.setColor(
-                ResourcesCompat.getColor(
-                    resources,
-                    R.color.color_yellow, null
-                )
+                    ResourcesCompat.getColor(
```
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt [hunk 6]
```diff
@@ -102,65 +106,72 @@ class DrawingActivity : DaggerAppCompatActivity(), DrawView {
...
+                            resources,
```
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt [hunk 7]
```diff
@@ -102,65 +106,72 @@ class DrawingActivity : DaggerAppCompatActivity(), DrawView {
...
+                            R.color.color_yellow, null
```
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt [hunk 8]
```diff
@@ -102,65 +106,72 @@ class DrawingActivity : DaggerAppCompatActivity(), DrawView {
...
+                    )
```
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt [hunk 9]
```diff
@@ -102,65 +106,72 @@ class DrawingActivity : DaggerAppCompatActivity(), DrawView {
...
             scaleColorView(image_color_yellow)
         }
 
         image_color_green.setOnClickListener {
             custom_draw_view.setColor(
-                ResourcesCompat.getColor(
-                    resources,
-                    R.color.color_green, null
-                )
+                    ResourcesCompat.getColor(
```
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt [hunk 10]
```diff
@@ -102,65 +106,72 @@ class DrawingActivity : DaggerAppCompatActivity(), DrawView {
...
+                            resources,
```
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt [hunk 11]
```diff
@@ -102,65 +106,72 @@ class DrawingActivity : DaggerAppCompatActivity(), DrawView {
...
+                            R.color.color_green, null
```
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt [hunk 12]
```diff
@@ -102,65 +106,72 @@ class DrawingActivity : DaggerAppCompatActivity(), DrawView {
...
+                    )
```
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt [hunk 13]
```diff
@@ -102,65 +106,72 @@ class DrawingActivity : DaggerAppCompatActivity(), DrawView {
...
             )
             scaleColorView(image_color_green)
         }
 
         image_color_blue.setOnClickListener {
             custom_draw_view.setColor(
-                ResourcesCompat.getColor(resources, R.color.color_blue, null)
+                    ResourcesCompat.getColor(resources, R.color.color_blue, null)
```
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt [hunk 14]
```diff
@@ -102,65 +106,72 @@ class DrawingActivity : DaggerAppCompatActivity(), DrawView {
...
             scaleColorView(image_color_blue)
         }
 
         image_color_pink.setOnClickListener {
             custom_draw_view.setColor(
-                ResourcesCompat.getColor(
-                    resources,
-                    R.color.color_pink, null
-                )
+                    ResourcesCompat.getColor(
```
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt [hunk 15]
```diff
@@ -102,65 +106,72 @@ class DrawingActivity : DaggerAppCompatActivity(), DrawView {
...
+                            resources,
```
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt [hunk 16]
```diff
@@ -102,65 +106,72 @@ class DrawingActivity : DaggerAppCompatActivity(), DrawView {
...
+                            R.color.color_pink, null
```
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt [hunk 17]
```diff
@@ -102,65 +106,72 @@ class DrawingActivity : DaggerAppCompatActivity(), DrawView {
...
+                    )
```
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt [hunk 18]
```diff
@@ -102,65 +106,72 @@ class DrawingActivity : DaggerAppCompatActivity(), DrawView {
...
             scaleColorView(image_color_pink)
         }
 
         image_color_brown.setOnClickListener {
             custom_draw_view.setColor(
-                ResourcesCompat.getColor(
-                    resources,
-                    R.color.color_brown, null
-                )
+                    ResourcesCompat.getColor(
```
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt [hunk 19]
```diff
@@ -102,65 +106,72 @@ class DrawingActivity : DaggerAppCompatActivity(), DrawView {
...
+                            resources,
```
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt [hunk 20]
```diff
@@ -102,65 +106,72 @@ class DrawingActivity : DaggerAppCompatActivity(), DrawView {
...
+                            R.color.color_brown, null
```
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt [hunk 21]
```diff
@@ -102,65 +106,72 @@ class DrawingActivity : DaggerAppCompatActivity(), DrawView {
...
+                    )
```
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt (hunk 17 above)
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt (hunk 16 above)
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt (hunk 15 above)
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt (hunk 16 above)
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt (hunk 17 above)
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt (hunk 18 above)
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt (hunk 19 above)
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt (hunk 20 above)
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt]: Wrong formatting.

**Review by philipbrito — CHANGES_REQUESTED** [review]

**Comment by kareemhamdy100** [comment]
@filipedelimabrito  I change the code format and code style 
what about the functionality is it good ?

**Comment by divyanshub024** [comment]
@kareemhamdy100 I see that you have added the `edit` button to show the drawing toolbar. It's better to hide the toolbar while drawing and show when you stop drawing. i.e when you lift your finger.

**Comment by kareemhamdy100** [comment]
@divyanshub024 I thinked in that way but found out that when lift your finger you can't see what you draw 
But I will try again

File: draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt [hunk 22]
```diff
@@ -44,7 +44,15 @@ class DrawingActivity : DaggerAppCompatActivity(), DrawView {
             .show()
     }
 
+
     private fun setupListeners() {
+        custom_draw_view.setOnTouchListener { view, event ->
+            custom_draw_view.onTouch(
+                event,
+                draw_tools,
+                ::toggleCompleteDrawTools)
```
Comment (philipbrito) [inline draw/src/main/java/chat/rocket/android/draw/main/ui/DrawActivity.kt:53]: `::toggleCompleteDrawTools` 👌 👏

**Review by philipbrito — APPROVED** [review]
I have fixed some issues with the code formatting here.
Thank you very much for sending this PR @kareemhamdy100 
💯

**Pull request merged**